drake/systems/plants/KinematicsCache.h

```diff
@@ -56,6 +56,8 @@ class KinematicsCache
 {
 private:
   std::unordered_map<RigidBody const *, KinematicsCacheElement<Scalar>, std::hash<RigidBody const *>, std::equal_to<RigidBody const *>, Eigen::aligned_allocator<std::pair<RigidBody const* const, KinematicsCacheElement<Scalar> > > > elements;
+  const int num_positions;
+  const int num_velocities;
```

Comment:
We have to store `num_velocities` in the KinematicsCache so that positionDotToVelocityMapping knows what size matrix to use, even if the KinematicsCache it belongs to wasn't created with velocities.